node/cn/filters/api.go

```diff
@@ -370,15 +370,13 @@ func (api *PublicFilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([
 		filter = NewBlockFilter(api.backend, *crit.BlockHash, crit.Addresses, crit.Topics)
 	} else {
 		// Convert the RPC block numbers into internal representations
-		var (
-			begin int64
-			end   int64
-		)
-		if crit.FromBlock == nil {
-			begin = int64(rpc.LatestBlockNumber)
+		begin := rpc.LatestBlockNumber.Int64()
```

Comment:
In Filter.Logs, if begin and end is -1, then set to header.Number.Uint64(). So the initial or undefined value of begin and end is -1. 
If header.Number.Uint64() is same as rpc.LatestBlockNumber.Int64(), it is ok.

Reply:
Yoops. rpc.LatestBlockNumber.Int64() is -1, so It's same as my intention.

Reply:
`-1` means `LatestBlockNumber`. So if the `begin` or `end` is -1, It will be set by `real latest block number` in `Filter`